File: backend/app/schemas/cron.py
"""Pydantic schemas for the Cron / Scheduled-task endpoints."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CronJobResponse(BaseModel):
    id: str
    name: str
    schedule: str
    enabled: bool
    last_run: Optional[datetime] = None
    last_status: Optional[str] = None
    next_run: Optional[datetime] = None


class CronExecutionResponse(BaseModel):
    id: int
    cron_job_id: str
    cron_job_name: Optional[str] = None
    status: str
    result: Optional[str] = None
    error_message: Optional[str] = None
    executed_at: datetime

    class Config:
        from_attributes = True


class CronRunResponse(BaseModel):
    success: bool
    message: str

File: frontend/src/api/cron.ts
import api from './index'

export interface CronJob {
  id: string
  name: string
  schedule: string
  enabled: boolean
  last_run?: string | null
  status?: string | null
}

export interface CronExecution {
  id: number
  cron_job_id: string
  status: string
  result?: string
  error_message?: string
  executed_at: string
}

export const cronApi = {
  listJobs() {
    return api.get<any, CronJob[]>('/api/cron/jobs/list')
  },
  runJob(jobId: string) {
    return api.post<any, { success: boolean; message: string }>(`/api/cron/jobs/${jobId}/run`)
  },
  listExecutions(jobId?: string) {
    return api.get<any, CronExecution[]>('/api/cron/executions', { params: { job_id: jobId } })
  }
}
